bundles/org.openhab.binding.nuki/src/main/java/org/openhab/binding/nuki/internal/handler/AbstractNukiDeviceHandler.java

```diff
@@ -92,23 +97,24 @@ protected void withHttpClient(Consumer<NukiHttpClient> consumer) {
         }
     }
 
-    public AbstractNukiDeviceHandler(Thing thing) {
-        super(thing);
-        this.configuration = getConfigAs(getConfigurationClass());
+    protected T loadConfiguration() {
+        T configuration = getConfigAs(getConfigurationClass());
         // legacy support - check if nukiId is hexadecimal (which might have been set by previous binding version)
         // and convert it to decimal
-        if (NUKI_ID_HEX_PATTERN.matcher(this.configuration.nukiId).matches()) {
+        if (NUKI_ID_HEX_PATTERN.matcher(configuration.nukiId).matches()) {
             logger.warn(
                     "SmartLock '{}' was created by old version of binding. It is recommended to delete it and discover again",
                     thing.getUID());
             this.thing.getConfiguration().put(NukiBindingConstants.PROPERTY_NUKI_ID,
```

Comment:
It looks like a non standard (and unexpected) way to update the thing configuration. Can you please fix that?

Reply:
https://www.openhab.org/docs/developer/bindings/#updating-the-configuration

Reply:
Ok, I've updated it, but now I'm getting another warning in log since this is done before the thing is initialized:
```
19:29:56.486 [WARN ] [internal.handler.NukiSmartLockHandler] - SmartLock 'nuki:smartlock:4c89efd66d:647771828' was created by old version of binding. It is recommended to delete it and discover again
19:30:43.432 [WARN ] [b.core.thing.binding.BaseThingHandler] - Handler NukiSmartLockHandler tried updating its configuration although the handler was already disposed.
```

Reply:
Yes, you should init configuration only in intialize, not in the constructor. Or at least you should not call loadConfiguration in the constructor because it updates the thing configuration too early, when not expected. 
Just call loadConfiguration in initialize and it will work.

Reply:
I was trying to avoid marking that `configuration` field as @Nullable since it's massive pain to deal with it (putting if != null every time I want to read it makes the code unreadable really fast), and if I don't initialize it in constructor it throws error.

Reply:
Yes you can do it but your method loadConfiguration is inappropriate for that. Just keep the initialize of the variable. 
And the stuff about updating the thing configuration, you do it only on initialize().

Reply:
Fixed, guess marking it as nullable does not really throw errors elsewhere